### PR TITLE
rbac-gen: updates for status

### PR DESCRIPTION
**Pull request by justinsb**
Add list & watch to the generated permissions (we already have get).

Also include permissions for querying the status subresource.

### CODE REVIEW

**Comment by atoato88** [comment]
Update codebase is OK for me, but CI test failed because convert logic is updated.
It's need to update [some golden tests](https://github.com/kubernetes-sigs/cluster-addons/tree/master/tools/rbac-gen/pkg/convert/tests), I think.

**Comment by justinsb** [comment]
I think the remaining failures are due to a change in behaviour of `go get` - sent #115 to (hopefully) fix

File: tools/rbac-gen/pkg/convert/tests/duplicates.out.yaml [hunk 1]
```diff
@@ -12,7 +12,18 @@ rules:
...
   - delete
   - get
+  - list
+  - patch
   - update
+  - watch
+- apiGroups:
+  - ""
+  resources:
+  - configmaps/status
```
Comment (atoato88) [inline tools/rbac-gen/pkg/convert/tests/duplicates.out.yaml:22]: I think configmaps don't have status subresource.
Should we update to generate only resources which has status subresource(e.g. deployments...), or set as future work?
Both is ok for me. :smile:

**Comment by atoato88** [comment]
@justinsb 
I left comment, please check.

**Comment by atoato88** [comment]
Can anyone check and approve?
I don't have approver role on this repo.

**Comment by atoato88** [comment]
/lgtm
I have not received some comments, but it's a small item, almost all items are OK.
I think it's good to merge this.
Sorry for my late decision.

**Pull request merged**